src/ssmDocument/commands/publishDocument.ts

```diff
@@ -0,0 +1,184 @@
+            localize(
+                'AWS.message.info.ssmDocument.publishDocument.createSuccess',
+                "Successfully created and uploaded SSM Document '{0}'",
+                wizardResponse.createResponse!.name
+            )
+        )
+        if (result.DocumentDescription) {
+            outputChannel.appendLine(stringify(result.DocumentDescription))
+        }
+
+        telemetry.recordSsmPublishDocumentCreate({ result: 'Succeeded' })
+        logger.info(`Created SSM Document successfully ${stringify(result.DocumentDescription)}`)
+        vscode.window.showInformationMessage(`Created SSM Document successfully`)
+        outputChannel.appendLine('')
+    } catch (err) {
+        outputChannel.appendLine(
+            localize(
+                'AWS.message.info.ssmDocument.publishDocument.createFailure',
+                "There was an error creating and uploading SSM Document '{0}', check logs for more information.",
+                wizardResponse.createResponse!.name
+            )
+        )
+
+        telemetry.recordSsmPublishDocumentCreate({ result: 'Failed' })
+        logger.info(`Failed to create SSM Document '${wizardResponse.createResponse!.name}'. %0`, err as Error)
+        vscode.window.showErrorMessage(
+            `Failed to create SSM Document '${wizardResponse.createResponse!.name}'. ${(err as Error).message}`
+        )
+        outputChannel.appendLine('')
+    }
+}
+
+export async function updateDocument(
+    wizardResponse: PublishSSMDocumentWizardResponse,
+    textDocument: vscode.TextDocument,
+    outputChannel: vscode.OutputChannel,
+    region: string,
+    client: SsmDocumentClient
+) {
+    const logger: Logger = getLogger()
+    logger.info(`Updating SSM Document '${wizardResponse.updateResponse!.name}'`)
+    outputChannel.show()
+    outputChannel.appendLine(
+        localize(
+            'AWS.message.info.ssmDocument.publishDocument.updating',
+            "Updating SSM Document '{0}' in {1}...",
+            wizardResponse.updateResponse!.name,
+            region
+        )
+    )
+
+    try {
+        const request: SSM.UpdateDocumentRequest = {
+            Content: textDocument.getText(),
+            Name: wizardResponse.updateResponse!.name,
+            DocumentVersion: '$LATEST',
+            DocumentFormat: textDocument.languageId === 'ssm-yaml' ? 'YAML' : 'JSON',
+        }
+
+        // Add more to request
+        const result = await client.updateDocument(request)
+        outputChannel.appendLine(
+            localize(
+                'AWS.message.info.ssmDocument.publishDocument.updateSuccess',
+                "Successfully updated SSM Document '{0}'",
+                wizardResponse.updateResponse!.name
+            )
+        )
+        if (result.DocumentDescription) {
+            outputChannel.appendLine(stringify(result.DocumentDescription))
+        }
+
+        telemetry.recordSsmPublishDocumentUpdate({ result: 'Succeeded' })
```

Comment:
Nit: our typical pattern for telemetry is to set the result at the start of the function as a `let` and call this once at the end of the action with the result potentially modified. In this case, maybe stick this in a `finally` statement and just set the result value in the `catch`? You already did something similar to this here: https://github.com/aws/aws-toolkit-vscode/pull/1288/files#diff-6c1ce33f48f364a936e2aa7e9e9c14bfR57